topology/graph/client.go

```diff
@@ -47,6 +47,8 @@ type AsyncClient struct {
 	Port      int
 	Path      string
 	messages  chan string
+	quit      chan bool
+	wg        *sync.WaitGroup
```

Comment:
Don't set a pointer here, use it directly as it not coming from outside the object
It will be auto instanciated with object